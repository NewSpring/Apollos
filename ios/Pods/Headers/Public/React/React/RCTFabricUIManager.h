../../../../../../node_modules/react-native/React/Fabric/RCTFabricUIManager.h