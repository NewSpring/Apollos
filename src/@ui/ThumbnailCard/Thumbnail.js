import { StyleSheet } from 'react-native';
import { compose } from 'recompose';

import styled from '@ui/styled';
import ProgressiveImage from '@ui/ProgressiveImage';
import { getIsLoading } from '@ui/isLoading';

const Image = compose(
  getIsLoading,
  styled({
    ...StyleSheet.absoluteFillObject,
    height: '100%',
    paddingTop: 0,
    alignItems: 'center',
    justifyContent: 'center',
  }, 'Card.Image'),
)(ProgressiveImage);

Image.propTypes = ProgressiveImage.propTypes;

export default Image;
