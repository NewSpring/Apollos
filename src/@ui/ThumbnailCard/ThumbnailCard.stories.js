import React from 'react';
import { View } from 'react-native';
import { storiesOf } from '@storybook/react-native';

import ThumbnailCard from './';

storiesOf('@ui/ThumbnailCard', module)
  .add('Default', () => {
    const container = {
      flex: 1,
      justifyContent: 'center',
      backgroundColor: '#f7f7f7',
    };

    return (
      <View style={container}>
        <ThumbnailCard
          title={'Why Jesus is Timeless'}
          image={'https://picsum.photos/400/400/?random'}
        />
      </View>
    );
  })
  .add('With Category', () => {
    const container = {
      flex: 1,
      justifyContent: 'center',
      backgroundColor: '#f7f7f7',
    };

    return (
      <View style={container}>
        <ThumbnailCard
          title={'Why Jesus is Timeless'}
          image={'https://picsum.photos/400/400/?random'}
          category={'Time Travel'}
        />
      </View>
    );
  })
  .add('Skeleton', () => {
    const container = {
      flex: 1,
      justifyContent: 'center',
      backgroundColor: '#f7f7f7',
    };

    return (
      <View style={container}>
        <ThumbnailCard
          title={'Why Jesus is Timeless'}
          image={'https://picsum.photos/400/400/?random'}
          category={'Time Travel'}
          isLoading
        />
      </View>
    );
  });
