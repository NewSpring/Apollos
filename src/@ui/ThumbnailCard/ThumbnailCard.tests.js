import React from 'react';
import renderer from 'react-test-renderer';
import Providers from '@ui/TestProviders';

import ThumbnailCard from './';

describe('the ThumbnailCard component', () => {
  it('should render', () => {
    const tree = renderer.create(
      <Providers>
        <ThumbnailCard
          title={'Why Jesus is Timeless'}
        />
      </Providers>,
    );
    expect(tree).toMatchSnapshot();
  });
  it('should render with an images', () => {
    const tree = renderer.create(
      <Providers>
        <ThumbnailCard
          title={'Why Jesus is Timeless'}
          images={'https://picsum.photos/100/100/?random'}
        />
      </Providers>,
    );
    expect(tree).toMatchSnapshot();
  });
  it('should render with a category', () => {
    const tree = renderer.create(
      <Providers>
        <ThumbnailCard
          title={'Why Jesus is Timeless'}
          category={'Time Travel'}
        />
      </Providers>,
    );
    expect(tree).toMatchSnapshot();
  });
  it('should render with a category and images', () => {
    const tree = renderer.create(
      <Providers>
        <ThumbnailCard
          title={'Why Jesus is Timeless'}
          images={'https://picsum.photos/100/100/?random'}
          category={'Time Travel'}
        />
      </Providers>,
    );
    expect(tree).toMatchSnapshot();
  });
  it('should render a loading state', () => {
    const tree = renderer.create(
      <Providers>
        <ThumbnailCard
          title={'Why Jesus is Timeless'}
          images={'https://picsum.photos/100/100/?random'}
          category={'Time Travel'}
          isLoading
        />
      </Providers>,
    );
    expect(tree).toMatchSnapshot();
  });
});
