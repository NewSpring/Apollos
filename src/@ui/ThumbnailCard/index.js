import React from 'react';
import PropTypes from 'prop-types';
import { compose, pure, setPropTypes } from 'recompose';
import { startCase, toLower } from 'lodash';

import { withIsLoading } from '@ui/isLoading';
import styled from '@ui/styled';
import Card, { CardContent } from '@ui/Card';
import SideBySideView from '@ui/SideBySideView';
import FlexedView from '@ui/FlexedView';
import { H5, BodyText } from '@ui/typography';
import Paragraph from '@ui/Paragraph';
import { enhancer as mediaQuery } from '@ui/MediaQuery';
import CategoryLabel from '@ui/CategoryLabel';

import Thumbnail from './Thumbnail';

const enhance = compose(
  setPropTypes({
    title: PropTypes.string.isRequired,
    description: PropTypes.oneOfType([PropTypes.string, PropTypes.node]),
    images: Thumbnail.propTypes.source,
    category: PropTypes.string,
    isLoading: PropTypes.bool,
  }),
  withIsLoading,
  pure,
);

const HorizontalLayout = styled({
  alignItems: 'center',
  minHeight: 110, // kind of the best middle ground for various title lengths.
})(SideBySideView);

const LeftColumn = compose(
  styled({ flex: 1.66 }),
  mediaQuery(
    ({ md }) => ({ maxWidth: md }),
    styled(({ theme }) => ({
      paddingVertical: theme.sizing.baseUnit * 0.75,
    })),
    styled(({ theme }) => ({
      paddingVertical: theme.sizing.baseUnit * 2,
      paddingHorizontal: theme.sizing.baseUnit * 1.5,
    })),
  ),
)(CardContent);

const RightColumn = styled({
  alignSelf: 'stretch',
})(FlexedView);

const ThumbnailCard = enhance(
  ({
    title, description, images, thumbnailImage, category, isLoading, ...otherProps
  }) => (
    <Card isLoading={isLoading} {...otherProps}>
      <HorizontalLayout>
        <LeftColumn>
          <H5>{title}</H5>
          {description ? (
            <Paragraph>
              <BodyText>{description}</BodyText>
            </Paragraph>
          ) : null}
          {typeof category !== 'undefined' ? (
            <CategoryLabel label={startCase(toLower(category))} isLoading={isLoading} />
          ) : null}
        </LeftColumn>
        {images ? (
          <RightColumn>
            <Thumbnail source={images} thumbnail={thumbnailImage} />
          </RightColumn>
        ) : null}
      </HorizontalLayout>
    </Card>
  ),
);

export default ThumbnailCard;
