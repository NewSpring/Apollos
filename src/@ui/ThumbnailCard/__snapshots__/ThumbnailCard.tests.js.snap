// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the ThumbnailCard component should render 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#ffffff",
      "borderRadius": 6,
      "marginHorizontal": 10,
      "marginVertical": 5,
      "shadowColor": "rgba(133, 133, 133, 0.30000000000000004)",
      "shadowOffset": Object {
        "height": 1,
        "width": 0,
      },
      "shadowOpacity": 1,
      "shadowRadius": 3,
    }
  }
>
  <View
    style={
      Object {
        "borderRadius": 6,
        "overflow": "hidden",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "flexWrap": "nowrap",
          "justifyContent": "space-between",
          "minHeight": 110,
        }
      }
    >
      <View
        style={
          Object {
            "flex": 1.66,
            "paddingHorizontal": 20,
            "paddingVertical": 15,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#303030",
              "fontFamily": "Colfax-Bold",
              "fontSize": 20.02,
              "lineHeight": 25.33,
              "paddingBottom": 3.15,
              "paddingTop": 4.5,
            }
          }
        >
          Why Jesus is Timeless
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`the ThumbnailCard component should render a loading state 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#ffffff",
      "borderRadius": 6,
      "marginHorizontal": 10,
      "marginVertical": 5,
      "shadowColor": "rgba(133, 133, 133, 0.30000000000000004)",
      "shadowOffset": Object {
        "height": 1,
        "width": 0,
      },
      "shadowOpacity": 1,
      "shadowRadius": 3,
    }
  }
>
  <View
    style={
      Object {
        "borderRadius": 6,
        "overflow": "hidden",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "flexWrap": "nowrap",
          "justifyContent": "space-between",
          "minHeight": 110,
        }
      }
    >
      <View
        style={
          Object {
            "flex": 1.66,
            "paddingHorizontal": 20,
            "paddingVertical": 15,
          }
        }
      >
        <View
          style={
            Object {
              "alignSelf": "stretch",
              "backgroundColor": "#dddddd",
              "borderRadius": 6,
              "height": 20.02,
              "marginVertical": 2.6549999999999994,
              "width": "60%",
            }
          }
        />
        <View
          flexed={false}
          style={
            Object {
              "alignItems": "center",
              "flex": null,
              "flexDirection": "row",
            }
          }
        >
          <View
            style={
              Object {
                "backgroundColor": "#dddddd",
                "borderRadius": 6,
                "height": 21.6,
                "width": 21.6,
              }
            }
          />
          <View
            style={
              Object {
                "flex": 1,
                "paddingHorizontal": 10,
              }
            }
          >
            <View
              style={
                Object {
                  "alignSelf": "stretch",
                  "backgroundColor": "#dddddd",
                  "borderRadius": 6,
                  "height": 14,
                  "marginVertical": 0.8899999999999997,
                  "width": "40%",
                }
              }
            />
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "alignSelf": "stretch",
            "flex": 1,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "aspectRatio": 1,
              "backgroundColor": "#dddddd",
              "bottom": 0,
              "height": "100%",
              "justifyContent": "center",
              "left": 0,
              "paddingTop": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
              "width": "100%",
            }
          }
        >
          <View
            style={
              Object {
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
              }
            }
          >
            <View
              style={
                Object {
                  "aspectRatio": 1,
                  "backgroundColor": "#dddddd",
                  "width": "100%",
                }
              }
            />
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`the ThumbnailCard component should render with a category 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#ffffff",
      "borderRadius": 6,
      "marginHorizontal": 10,
      "marginVertical": 5,
      "shadowColor": "rgba(133, 133, 133, 0.30000000000000004)",
      "shadowOffset": Object {
        "height": 1,
        "width": 0,
      },
      "shadowOpacity": 1,
      "shadowRadius": 3,
    }
  }
>
  <View
    style={
      Object {
        "borderRadius": 6,
        "overflow": "hidden",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "flexWrap": "nowrap",
          "justifyContent": "space-between",
          "minHeight": 110,
        }
      }
    >
      <View
        style={
          Object {
            "flex": 1.66,
            "paddingHorizontal": 20,
            "paddingVertical": 15,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#303030",
              "fontFamily": "Colfax-Bold",
              "fontSize": 20.02,
              "lineHeight": 25.33,
              "paddingBottom": 3.15,
              "paddingTop": 4.5,
            }
          }
        >
          Why Jesus is Timeless
        </Text>
        <View
          flexed={false}
          style={
            Object {
              "alignItems": "center",
              "flex": null,
              "flexDirection": "row",
            }
          }
        >
          <Svg
            height={21.6}
            viewBox="0 0 24 24"
            width={21.6}
          >
            <Path
              d="M4.6 3.64v16.356c0 .004 0 .004-.003.004h14.806c-.004 0-.003 0-.003-.004V3.64c0-.004 0-.004.003-.004H4.597c.004 0 .003 0 .003.004zm-1.6 0C3 2.734 3.717 2 4.597 2h14.806C20.285 2 21 2.734 21 3.64v16.356c0 .906-.717 1.64-1.597 1.64H4.597c-.882 0-1.597-.734-1.597-1.64V3.64zm4 3.27c0 .45.366.817.818.817h8.364c.452 0 .818-.366.818-.818 0-.454-.366-.82-.818-.82H7.818c-.452 0-.818.366-.818.82zm0 4.908c0 .452.366.818.818.818h8.364c.452 0 .818-.366.818-.818 0-.452-.366-.818-.818-.818H7.818c-.452 0-.818.366-.818.818zm0 4.91c0 .45.366.817.818.817h4.764c.452 0 .818-.366.818-.818 0-.452-.366-.818-.818-.818H7.818c-.452 0-.818.364-.818.816z"
              fill="#505050"
            />
          </Svg>
          <View
            style={
              Object {
                "flex": 1,
                "paddingHorizontal": 10,
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "#505050",
                  "fontFamily": "Colfax-Regular",
                  "fontSize": 14,
                  "lineHeight": 15.78,
                  "paddingBottom": 1.8,
                  "paddingTop": 3.6,
                }
              }
            >
              Time Travel
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`the ThumbnailCard component should render with a category and images 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#ffffff",
      "borderRadius": 6,
      "marginHorizontal": 10,
      "marginVertical": 5,
      "shadowColor": "rgba(133, 133, 133, 0.30000000000000004)",
      "shadowOffset": Object {
        "height": 1,
        "width": 0,
      },
      "shadowOpacity": 1,
      "shadowRadius": 3,
    }
  }
>
  <View
    style={
      Object {
        "borderRadius": 6,
        "overflow": "hidden",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "flexWrap": "nowrap",
          "justifyContent": "space-between",
          "minHeight": 110,
        }
      }
    >
      <View
        style={
          Object {
            "flex": 1.66,
            "paddingHorizontal": 20,
            "paddingVertical": 15,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#303030",
              "fontFamily": "Colfax-Bold",
              "fontSize": 20.02,
              "lineHeight": 25.33,
              "paddingBottom": 3.15,
              "paddingTop": 4.5,
            }
          }
        >
          Why Jesus is Timeless
        </Text>
        <View
          flexed={false}
          style={
            Object {
              "alignItems": "center",
              "flex": null,
              "flexDirection": "row",
            }
          }
        >
          <Svg
            height={21.6}
            viewBox="0 0 24 24"
            width={21.6}
          >
            <Path
              d="M4.6 3.64v16.356c0 .004 0 .004-.003.004h14.806c-.004 0-.003 0-.003-.004V3.64c0-.004 0-.004.003-.004H4.597c.004 0 .003 0 .003.004zm-1.6 0C3 2.734 3.717 2 4.597 2h14.806C20.285 2 21 2.734 21 3.64v16.356c0 .906-.717 1.64-1.597 1.64H4.597c-.882 0-1.597-.734-1.597-1.64V3.64zm4 3.27c0 .45.366.817.818.817h8.364c.452 0 .818-.366.818-.818 0-.454-.366-.82-.818-.82H7.818c-.452 0-.818.366-.818.82zm0 4.908c0 .452.366.818.818.818h8.364c.452 0 .818-.366.818-.818 0-.452-.366-.818-.818-.818H7.818c-.452 0-.818.366-.818.818zm0 4.91c0 .45.366.817.818.817h4.764c.452 0 .818-.366.818-.818 0-.452-.366-.818-.818-.818H7.818c-.452 0-.818.364-.818.816z"
              fill="#505050"
            />
          </Svg>
          <View
            style={
              Object {
                "flex": 1,
                "paddingHorizontal": 10,
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "#505050",
                  "fontFamily": "Colfax-Regular",
                  "fontSize": 14,
                  "lineHeight": 15.78,
                  "paddingBottom": 1.8,
                  "paddingTop": 3.6,
                }
              }
            >
              Time Travel
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "alignSelf": "stretch",
            "flex": 1,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "aspectRatio": 1,
              "backgroundColor": "#dddddd",
              "bottom": 0,
              "height": "100%",
              "justifyContent": "center",
              "left": 0,
              "paddingTop": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
              "width": "100%",
            }
          }
        >
          <View
            style={
              Object {
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
              }
            }
          >
            <View
              style={
                Object {
                  "aspectRatio": 1,
                  "backgroundColor": "#dddddd",
                  "width": "100%",
                }
              }
            />
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`the ThumbnailCard component should render with a description 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#ffffff",
      "borderRadius": 6,
      "marginHorizontal": 10,
      "marginVertical": 5,
      "shadowColor": "rgba(133, 133, 133, 0.30000000000000004)",
      "shadowOffset": Object {
        "height": 1,
        "width": 0,
      },
      "shadowOpacity": 1,
      "shadowRadius": 3,
    }
  }
>
  <View
    style={
      Object {
        "borderRadius": 6,
        "overflow": "hidden",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "flexWrap": "nowrap",
          "justifyContent": "space-between",
          "minHeight": 110,
        }
      }
    >
      <View
        style={
          Object {
            "flex": 1.66,
            "paddingHorizontal": 20,
            "paddingVertical": 15,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#303030",
              "fontFamily": "Colfax-Bold",
              "fontSize": 20.02,
              "lineHeight": 25.33,
              "paddingBottom": 3.15,
              "paddingTop": 4.5,
            }
          }
        >
          Why Jesus is Timeless
        </Text>
        <View
          style={
            Object {
              "paddingVertical": 10,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            bold={false}
            ellipsizeMode="tail"
            italic={false}
            style={
              Object {
                "color": "#303030",
                "fontFamily": "DroidSerif-Regular",
                "fontSize": 20.02,
                "lineHeight": 33.11,
                "paddingBottom": 2.25,
                "paddingTop": 1.8,
              }
            }
          >
            This is the story of a God, who made a flood and drowned the whole world. He gave up His son for everybody, and we absolutely love Him, all the time.
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`the ThumbnailCard component should render with an images 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#ffffff",
      "borderRadius": 6,
      "marginHorizontal": 10,
      "marginVertical": 5,
      "shadowColor": "rgba(133, 133, 133, 0.30000000000000004)",
      "shadowOffset": Object {
        "height": 1,
        "width": 0,
      },
      "shadowOpacity": 1,
      "shadowRadius": 3,
    }
  }
>
  <View
    style={
      Object {
        "borderRadius": 6,
        "overflow": "hidden",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "flexWrap": "nowrap",
          "justifyContent": "space-between",
          "minHeight": 110,
        }
      }
    >
      <View
        style={
          Object {
            "flex": 1.66,
            "paddingHorizontal": 20,
            "paddingVertical": 15,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#303030",
              "fontFamily": "Colfax-Bold",
              "fontSize": 20.02,
              "lineHeight": 25.33,
              "paddingBottom": 3.15,
              "paddingTop": 4.5,
            }
          }
        >
          Why Jesus is Timeless
        </Text>
      </View>
      <View
        style={
          Object {
            "alignSelf": "stretch",
            "flex": 1,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "aspectRatio": 1,
              "backgroundColor": "#dddddd",
              "bottom": 0,
              "height": "100%",
              "justifyContent": "center",
              "left": 0,
              "paddingTop": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
              "width": "100%",
            }
          }
        >
          <View
            style={
              Object {
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
              }
            }
          >
            <View
              style={
                Object {
                  "aspectRatio": 1,
                  "backgroundColor": "#dddddd",
                  "width": "100%",
                }
              }
            />
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
